src/material-experimental/mdc-list/testing/list-harness-base.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ComponentHarness,
+  ComponentHarnessConstructor,
+  HarnessPredicate
+} from '@angular/cdk/testing';
+import {DividerHarnessFilters, MatDividerHarness} from '@angular/material/divider/testing';
+import {BaseListItemHarnessFilters, SubheaderHarnessFilters} from './list-harness-filters';
+import {MatSubheaderHarness} from './list-item-harness-base';
+
+/** Represents a section of a list falling under a specific header. */
+export interface ListSection<I> {
+  /** The heading for this list section. `undefined` if there is no heading. */
+  heading?: string;
+
+  /** The items in this list section. */
+  items: I[];
+}
+
+/**
+ * Shared behavior among the harnesses for the various `MatList` flavors.
+ * @template T A constructor type for a list item harness type used by this list harness.
+ * @template C The list item harness type that `T` constructs.
+ * @template F The filter type used filter list item harness of type `C`.
+ * @docs-private
+ */
+export class MatListHarnessBase<
```

Comment:
Should this be abstract?

Reply:
Yeah, sounds reasonable. This logic has been taken from the existing list harness, so I'll also update it there.